### ISSUE TITLE
0.61- 0.60 Beta 2 issues with keeping track of level progress in First Strike/Counterstrike 

### ISSUE THREAD

**Issue by Descender1032** [dxx-rebirth/dxx-rebirth#494]
I've had problems keeping track of level progress in Counterstrike. Right now it won't go past 22, whether a level is completed with cheats or not. Doesn't seem to be associated with the pilot file, as I created a fresh pilot profile, teleported to level 24 and completed it - the highest selectable level was still 22. Before reaching level 21, I do recall my progress counter actually resetting (was up to 21 one time, reverted back to the initial state later). Right now it seems stable, but fails to exceed level 22. On the contrary, custom missions like Vertigo seem to keep track of the levels just fine. 

I've got a similar problem with D1x - same build and version - it fails to keep track of my progress entirely for the main mission - I can only start from level 1 and beating the level doesn't give me access to level 2.

I'm using an altered HAM file and altered S11 and S22 libraries for the sound patch.

**Comment by vLKp** [comment]
`0.61 - 0.60 Beta 2` is unclear to me.  What version are you actually playing?  Commit 03b6e604ebf7780db8d827f23612f7787a51f6aa or something much older?

I cannot reproduce the reported problem, including using a new pilot, when running with the latest code on `Descent 2: Counterstrike`.

**Comment by Descender1032** [comment]
Hey, 

I downloaded the as-of-then latest package from the link provided on Discord, then replaced all my files, except for the ini file. That might have somethign to do with it - maybe the ini syntax has been updated between the versions?

**Comment by vLKp** [comment]
No, ini syntax has not changed in a long time.  Any incompatible changes there would likely be designed to provoke an error message, rather than silent incorrect results.

Since I don't know when you read a link off Discord, and I don't know any of the links that have been posted there, I don't know what you downloaded.  Please link directly to the program archive, as I quoted above.  Instructions that might (or might not, depending on changes over time) let me guess the link are not enough for me to know what you have.  Since I cannot reproduce the problem with the latest code, and I don't know what code you used, there is currently nothing I can do for you.

**Comment by Descender1032** [comment]
Well, it would be difficult to point directly to the archive, since it was downloaded using the generic download button here: https://www.dxx-rebirth.com/download-dxx-rebirth/

The archive I downloaded was updated on Discord o 27.12.2019. Haven't tried the January version yet, that may actually solve the issue.

**Comment by vLKp** [comment]
There are no download buttons on that page.  There are three download links, all of which can have their link target copied.  Currently, their values are:

- https://www.dxx-rebirth.com/download/dxx/rebirth/dxx-rebirth_20200116-src.tar.xz
- https://www.dxx-rebirth.com/download/dxx/rebirth/dxx-rebirth_20200116-win.zip
- https://www.dxx-rebirth.com/download/dxx/user/afuturepilot/download.php?f=DXX-Rebirth_Setup.exe

Do you mean that you used the `Third-party packages` entry for the Windows installer, which lacks any versioning in the filename?

What version does the game claim to be when you run it?

---

When you tried this, had you used any cheat codes?  I had an invalid issue report a while back because someone was using cheats and expected the game to save his progress.

**Comment by Descender1032** [comment]
No, it's always completely legit progress, no cheats involved. Then again, in older versions beating a level even after warping there via "Freespace" would save the mission's progress anyway. 

https://www.dxx-rebirth.com/download/dxx/rebirth/dxx-rebirth_20200116-win.zip This is the link I used - I actually went and got the newest version to stay up to date. Good news - I completed level 22 with no cheats on Trainee, and level 23 got unlocked. It would seem the current version resolves the problem.
